File: app/src/main/java/amrabed/android/release/evaluation/edit/drag/DragListener.java
package amrabed.android.release.evaluation.edit.drag;

import androidx.recyclerview.widget.RecyclerView.ViewHolder;

public interface DragListener {
    void onDrag(ViewHolder holder);

    void onItemMoved(ViewHolder source, ViewHolder destination);

    void onItemRemoved(ViewHolder holder);
}

File: app/src/main/java/amrabed/android/release/evaluation/edit/drag/DragListener.kt
package amrabed.android.release.evaluation.edit.drag

import androidx.recyclerview.widget.RecyclerView

interface DragListener {
    fun onDrag(holder: RecyclerView.ViewHolder?)
    fun onItemMoved(source: RecyclerView.ViewHolder, destination: RecyclerView.ViewHolder)
    fun onItemRemoved(holder: RecyclerView.ViewHolder)
    fun onItemHidden(holder: RecyclerView.ViewHolder)
    fun onItemAdded(position: Int)
}